lib/videoroom/room.ex

```diff
@@ -175,27 +202,33 @@ defmodule Videoroom.Room do
 
   @impl true
   def handle_info({:DOWN, _ref, :process, pid, _reason}, state) do
-    if pid == state.rtc_engine do
-      room_span_id(state.room_id)
-      |> Membrane.OpenTelemetry.end_span()
+    {peer_id, _peer_channel_id} =
+      state.peer_channels
+      |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
 
-      {:stop, :normal, state}
-    else
-      {peer_id, _peer_channel_id} =
-        state.peer_channels
-        |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
+    Membrane.Logger.info("Peer #{inspect(peer_id)} left")
 
-      Membrane.Logger.info("Peer #{inspect(peer_id)} left RTC Engine")
+    Engine.remove_endpoint(state.rtc_engine, peer_id)
+    {_elem, state} = pop_in(state, [:peer_channels, peer_id])
 
-      Engine.remove_endpoint(state.rtc_engine, peer_id)
-      {_elem, state} = pop_in(state, [:peer_channels, peer_id])
+    if state.peer_channels == %{} do
+      Membrane.Logger.info("Last peer left the room. Terminating engine.")
 
-      if state.peer_channels == %{} do
-        Engine.terminate(state.rtc_engine)
-        {:stop, :normal, state}
-      else
-        {:noreply, state}
+      case Engine.terminate(state.rtc_engine, blocking?: true) do
+        :ok ->
+          Membrane.Logger.info("Engine terminated.")
+          {:stop, :normal, state}
+
+        error ->
+          Membrane.Logger.error(
+            "Couldn't terminate engine gracefully: #{inspect(error)}. Forcing termination."
+          )
+
+          Process.exit(state.rtc_engine, :kill)
+          {:noreply, state}
```

Comment:
Shouldn't we do here?
```suggestion
          {:stop, :normal, state}
```